src/Xamarin.Android.Build.Tasks/Tasks/LinkAssembliesNoShrink.cs

```diff
@@ -53,10 +53,12 @@ public override bool RunTask ()
 				for (int i = 0; i < SourceFiles.Length; i++) {
 					var source = SourceFiles [i];
 					var destination = DestinationFiles [i];
+					var assemblyDefinition = resolver.GetAssembly (source.ItemSpec);
+
+					step.CheckAppDomainUsage (assemblyDefinition, (string msg) => Log.LogWarning (msg));
```

Comment:
As @jonathanpeppers mentioned earlier, this should be a `LogCodedWarning()`, and please:

1. Create a new warning code
2. Add docs for the new code in https://github.com/xamarin/xamarin-android/tree/master/Documentation/guides/messages

Perhaps `XA2000 member semantics will change in a future release` would work?